### PR TITLE
Split/rename util.c+h and def.h

### PR DESCRIPTION
**Pull request by keszybz**
This pull request gets rid of util.c+h, which were containing various completely unrelated utilities.
`basic/initrd-util.c+h`, `basic/argv-util.c+h`, `basic/logarithm.h`, along with `test-raw-clone.c`,
`test-argv-util.c`, `test-limits-util.c`, and `test-logarithm.c`, are new.

Also, `def.h` is renamed to `constants.h`.

No functional change is intended.

### CODE REVIEW

File: src/basic/raw-clone.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #include "log.h"
 #include "macro.h"
+#include "process-util.h"
```
Comment (bluca) [inline src/basic/raw-clone.h:14]: why is this added here rather than to the test?

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
```
../src/src/basic/argv-util.c: In function ‘update_argv’:
../src/src/basic/argv-util.c:90:13: error: implicit declaration of function ‘geteuid’; did you mean ‘getdelim’? [-Werror=implicit-function-declaration]
         if (geteuid() != 0)
```

File: src/basic/raw-clone.h (hunk 1 above)
Comment (keszybz) [reply src/basic/raw-clone.h:14]: Because `reset_cached_pid()` is called below. This essentially fixes a pre-existing error.

Comment (keszybz) [reply src/basic/raw-clone.h:14]: I added a comment to the commit message.

**Review by bluca — APPROVED** [review]

File: src/libsystemd/sd-journal/journal-file.c [hunk 2]
```diff
@@ -92,6 +92,19 @@
 #  pragma GCC diagnostic ignored "-Waddress-of-packed-member"
 #endif
 
+_const_ static int mmap_prot_from_open_flags(int flags) {
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-file.c]: `_const_` has no relevance for functions that are static anyway. `_const_` is a hint to the compiler that some function doesn't need to be called twice for the same value because it is side-effect free. The compiler can't know that if it looks at a public symbol, so we can tell it that. But, if the function is "static", then the compiler knows anyway if is const or not.

(Moreover, given you added assert_not_reached() here, it's not side-effect free anymore anyway, so it's not const either anymore)

**Comment by keszybz** [comment]
```
12:05:31     archlinux_systemd_ci: In file included from ../build/src/shared/service-util.c:7:
12:05:31     archlinux_systemd_ci: ../build/src/basic/build.h:4:10: fatal error: version.h: No such file or directory
12:05:31     archlinux_systemd_ci:     4 | #include "version.h"
12:05:31     archlinux_systemd_ci:       |          ^~~~~~~~~~~
12:05:31     archlinux_systemd_ci: compilation terminated.
```
Hmm.

I think it's either a meson error or something is misdeclared somewhere.

File: src/libsystemd/sd-journal/journal-file.c [hunk 3]
```diff
@@ -92,6 +92,19 @@
...
+        case O_RDONLY:
+                return PROT_READ;
+        case O_WRONLY:
+                return PROT_WRITE;
+        case O_RDWR:
+                return PROT_READ|PROT_WRITE;
+        default:
+                assert_not_reached();
+        }
+}
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-file.c:106]: this function is *super* generic btw, I'd probably keep it in the shared path, it sounds like this is what we want sooner or later for any mmap() code...

Comment (keszybz) [reply src/libsystemd/sd-journal/journal-file.c:106]: Hmm, it has exactly one user, added in 2011. mmap is not a terribly nice interface to use, and I think it's quite likely that we'll never add any more uses. But if that happens, it should be trivial to move the function to some public header… I don't think we need to do this proactively for an uncertain future use case.

File: src/libsystemd/sd-journal/journal-file.c (hunk 2 above)
Comment (keszybz) [reply src/libsystemd/sd-journal/journal-file.c]: I dropped the `_const_` now.

**Comment by keszybz** [comment]
I think it should pass now.

**Comment by bluca** [comment]
computer says no:

```
13:11:34     archlinux_systemd_ci: In file included from ../build/src/journal/journalctl.c:22:
13:11:34     archlinux_systemd_ci: ../build/src/basic/build.h:4:10: fatal error: version.h: No such file or directory
13:11:34     archlinux_systemd_ci:     4 | #include "version.h"
```

**Comment by dtardon** [comment]
RIP util.c (2009 - 2022)

**Comment by bluca** [comment]
> RIP util.c (2009 - 2022)

So long, and thanks for all the utils
